chore: Upgrade Pytest to `^7.1.2`

WillDaSilva opened the pull request:
Closes #3443

WillDaSilva committed:
Upgrade to Pytest v7 & misc other dep updates

WillDaSilva committed:
Use repr to interpolate config into source code string

WillDaSilva commented:
Looks like there are issues with coverage that'll be sorted out by #6304

WillDaSilva committed:
chore: Move Coverage config into `pyproject.toml` & update it (#6304)

WillDaSilva committed:
Merge branch 'main' into pytest-v7

edgarrmondragon reviewed (approved):
LGTM! You're bringing Meltano's dependencies to 2022 🚀

WillDaSilva committed:
Merge branch 'main' into pytest-v7

WillDaSilva committed:
Do not collect coverage in GitLab CI

Trying to collect coverage in GitLab CI is resulting in errors. Rather
than fixing them, we can simply remove that feature since we're
ultimately moving away from GitLab CI entirely, and we already get
coverage from GitHub CI.

WillDaSilva commented:
The decrease in reported coverage is a result of us now tracking branch coverage too. The coverage report has gotten more accurate, and revealed that our code coverage is not as good as previously reported.
